horizons/world/building/building.py

```diff
@@ -230,62 +230,21 @@ def getInstance(cls, session, x, y, action='idle', level=0, rotation=45, action_
 		#rotation = cls.check_build_rotation(session, rotation, x, y)
 		# TODO: replace this with new buildable api
 		# IDEA: save rotation in savegame
-		facing_loc = fife.Location(session.view.layers[cls.layer])
-		instance_coords = list((x, y, 0))
-		layer_coords = list((x, y, 0))
+
 		width, length = cls.size
+		if rotation == 135 or rotation == 315:
+			width, length = length, width
```

Comment:
I think this line needs a short comment explaining why `width` and `length` are switched.

Reply:
I already considered this but I couldn't think of anything that didn't say the same as the code itself